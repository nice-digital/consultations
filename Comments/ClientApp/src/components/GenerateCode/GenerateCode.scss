@use "@nice-digital/nds-core/scss/spacing";
@use "@nice-digital/nds-core/scss/utils";

.organisation-codes {
  &__row {
    display: flex;
    align-items: center;
    padding: utils.rem(spacing.$small);
  }

  &__name, &__generate {
    flex: 0 0 50%;
    display: flex;
    align-items: center;
    justify-content: space-between;

    p, button {
      margin: 0;
    }
  }
}

.btn {
  &.btn--copied {
    background-color: #e6faff;
    color: #000;
    padding: 0 utils.rem(spacing.$medium);
    margin-bottom: 0;
    border: none;

    &:focus, &:hover {
      background-color: #e6faff;
      color: #000;
    }
  }
}

.arrow-icon {
  &:after {
    display: inline-block;
    font-family: 'NICE Icons';
    content: '\e03c';
    font-size: 0.875rem;
    line-height: 1.25rem;
  }

  [aria-expanded='true'] & {
    &:after {
      content: '\e039';
    }
  }
}
