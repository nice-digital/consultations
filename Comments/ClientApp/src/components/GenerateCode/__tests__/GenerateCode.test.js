import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { GenerateCode } from "../GenerateCode";

const fakePropsNoCode = {
	organisationCodes: [{
		canGenerateCollationCode: true,
		collationCode: null,
		organisationAuthorisationId: 0,
		organisationId: 999,
		organisationName: "Super Magic Club",
	}],
	consultationId: 111,
};

test("should match snapshot with no code having been generated", () => {
	const {container} = render(<GenerateCode {...fakePropsNoCode} />);
	const shareOrganisationButton = screen.getByRole("button", { name: "Share with organisation" });
	fireEvent.click(shareOrganisationButton);
	expect(container).toMatchSnapshot();
});

test("should match snapshot with code having been generated", () => {
	let fakePropsCode = {
		organisationCodes: [{...fakePropsNoCode.organisationCodes[0]}],
		consultationId: 111,
	};
	fakePropsCode.organisationCodes[0].canGenerateCollationCode = false;
	fakePropsCode.organisationCodes[0].collationCode = "1234 5678 9123";
	const {container} = render(<GenerateCode {...fakePropsCode} />);
	const shareOrganisationButton = screen.getByRole("button", { name: "Share with organisation" });
	fireEvent.click(shareOrganisationButton);
	expect(container).toMatchSnapshot();
});

test("shouldn't show the panel when first loaded", () => {
	render(<GenerateCode {...fakePropsNoCode} />);
	const organisationCodesHeading = screen.queryAllByRole("heading", { name: "Generate a code to share the consultation" });
	expect(organisationCodesHeading.length).toBe(0);
});

test("should show panel when share button is clicked", () => {
	render(<GenerateCode {...fakePropsNoCode} />);
	const shareOrganisationButton = screen.getByRole("button", { name: "Share with organisation" });
	fireEvent.click(shareOrganisationButton);
	const organisationCodesHeading = screen.queryAllByRole("heading", { name: "Generate a code to share the consultation" });
	expect(organisationCodesHeading.length).toBe(1);
});
