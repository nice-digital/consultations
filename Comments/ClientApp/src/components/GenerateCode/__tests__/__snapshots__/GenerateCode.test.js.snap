// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with code having been generated 1`] = `
<div>
  <div>
    <button
      aria-controls="organisation-codes-111"
      aria-expanded="true"
      class="btn btn--inverse"
      data-qa-sel="share-with-org-button"
      id="share-organisation-111"
      type="button"
    >
      Share with organisation 
      <span
        class="arrow-icon"
      />
    </button>
    <div
      class="panel organisation-codes mv--b"
      id="organisation-codes-111"
    >
      <h2
        class="h5"
      >
        Generate a code to share the consultation
      </h2>
      <div
        class="organisation-codes__row"
      >
        <div
          class="organisation-codes__name"
        >
          <p>
            Super Magic Club
          </p>
        </div>
        <div
          class="organisation-codes__generate pl--b"
        >
          <span>
            1234 5678 9123
          </span>
          <button
            aria-live="polite"
            class="btn btn--copied"
            data-qa-sel="copy-code-button"
            type="button"
          >
            Copy code
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should match snapshot with no code having been generated 1`] = `
<div>
  <div>
    <button
      aria-controls="organisation-codes-111"
      aria-expanded="true"
      class="btn btn--inverse"
      data-qa-sel="share-with-org-button"
      id="share-organisation-111"
      type="button"
    >
      Share with organisation 
      <span
        class="arrow-icon"
      />
    </button>
    <div
      class="panel organisation-codes mv--b"
      id="organisation-codes-111"
    >
      <h2
        class="h5"
      >
        Generate a code to share the consultation
      </h2>
      <div
        class="organisation-codes__row"
      >
        <div
          class="organisation-codes__name"
        >
          <p>
            Super Magic Club
          </p>
        </div>
        <div
          class="organisation-codes__generate pl--b"
        >
          <button
            aria-live="polite"
            class="btn"
            data-qa-sel="generate-code-button"
            type="button"
          >
            Generate
          </button>
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
